documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -54,6 +54,20 @@ If there's no match within the field we want to highlight, be it because the doc
 of some other predicate condition, or because the highlighted field wasn't a part of a predicate at all, an empty list will be returned by default.
 See <<search-dsl-highlighting-highlighter-options-no-match-size,no match size configuration>> option for more details on how this can be adjusted.
 <5> Fetch the results, which will have highlighted fragments.
+As an example, the result may look like:
+----
+[
+    # Hit #0:
+    ["The Boscombe Valley <em>Mystery</em>"],
+    # Hit #1:
+    ["A Caribbean <em>Mystery</em>"],
+    # Hit #2:
+    ["A <em>Mystery</em> of <em>Mysteries</em>: The Death and Life of Edgar Allan Poe"]
+    # Other hits
+    ...
+]
+----
+Note that the result of applying a highlight projection is always a list of `String`.
```

Comment:
If you add a blank line before this, it will move the snippet out of the callout `<5>` and then you might be able to use callouts instead of comments:

```suggestion
Note that the result of applying a highlight projection is always a list of `String`.

As an example, the result may look like:

[source, JSON]
----
[
    ["The Boscombe Valley <em>Mystery</em>"], // <1>
    ["A Caribbean <em>Mystery</em>"], // <2>
    ["A <em>Mystery</em> of <em>Mysteries</em>: The Death and Life of Edgar Allan Poe"]  // <3>
]
----
<1> First hit.
<2> Second hit.
<3> Third hit.
```

(I used the JSON syntax highlighting here since we already did that [elsewhere](https://github.com/hibernate/hibernate-search/blob/26f1da6bef433ca33650d22b010ace005c745e91/documentation/src/main/asciidoc/reference/binding-valuebridge.asciidoc?plain=1#L73-L82))

Reply:
😃 
also, I've tried and from what it looks it's not necessary to have a  `//` in json callouts so I've used just <#>